pandas/core/indexes/base.py

```diff
@@ -1412,6 +1412,12 @@ def is_monotonic(self):
         """ alias for is_monotonic_increasing (deprecated) """
         return self.is_monotonic_increasing
 
+    @property
```

Comment:
are we actually using this? (it seems you in-lined it below), I actually prefer that to making a new method

Reply:
The names are similar, but I'm calling this function twice in the intersection function to help me deal with the boolean logic overloading. I can remove this, and put this inline if you really want it that way.